src/common/post.ts

```diff
@@ -329,3 +344,125 @@ export const insertCodeSnippetsFromUrl = async ({
     throw err;
   }
 };
+
+type ModeratedPostCdc = ChangeObject<SourcePostModeration>;
+
+export const updateModeratedPost = async (
+  con: ConnectionManager,
+  moderated: ModeratedPostCdc,
+): Promise<ModeratedPostCdc | null> => {
+  if (!moderated?.postId) {
+    logger.error('unable to update moderated post', { moderated });
+    return null;
+  }
+
+  const postParam = { id: moderated.postId };
+  const repo = con.getRepository(Post);
+
+  await repo.findOneOrFail({ where: { id: moderated.postId } });
+
+  const updatedPost: Partial<FreeformPost> = {
+    title: moderated.title,
+    titleHtml: moderated.titleHtml,
+  };
+
+  if (moderated.content) {
+    updatedPost.content = moderated.content;
+    updatedPost.contentHtml = moderated.contentHtml!;
+  }
+
+  if (moderated.image) {
+    updatedPost.image = moderated.image;
+  }
+
+  await repo.update(postParam, updatedPost);
+
+  return moderated;
+};
+
+export const getExistingPost = async (
```

Comment:
@rebelchris I agree with duplicated functions, but I think it just comes to the fact we often add utility functions in different files without really looking if they are used or if there is something similar. 

Would say this is an example, its used in a single place and for sure if I need to check existing post I'll just write it myself again.

Reply:
Yeah probably good idea, it's not blocking for now, we can log it for web guild meeting :) 
It's good to think about these systems to avoid re-building stuff.